src/components/Image.ts

```diff
@@ -179,21 +298,105 @@ export class Image extends Component<ImageProps, ImageChild> {
 			);
 		}
 
-		const blipEmbedRel = evaluateXPathToString(
-			`${QNS.pic}blipFill/${QNS.a}blip/@${QNS.r}embed/string()`,
-			picNode,
-		);
-		const location = relationships.getTarget(blipEmbedRel);
-		const data = archive.readBinary(location);
+		const blipNode = evaluateXPathToFirstNode(`${QNS.pic}blipFill/${QNS.a}blip`, picNode);
+		if (blipNode === null) {
+			throw new Error('Failed to load image. No blip found inside a blipFill.');
+		}
+		const { main, svg } = extractData(archive, relationships, blipNode);
+
+		const dataExtensions: DataExtensions = {};
+		if (svg) {
+			dataExtensions.svg = svg.data;
+		}
+
 		const image = new Image({
-			data,
+			data: main.data,
+			dataExtensions,
 			title,
 			width,
 			height,
 		});
-		image.#location = location;
+		image.#embedMeta.location = main.location;
+		if (svg) {
+			const { svg: svgEmbed } = image.#embedMeta.extensions;
+			if (!svgEmbed) {
```

Comment:
No need to handle this case specifically, an error like `Uncaught TypeError: Cannot set properties of undefined (setting 'location')` that is thrown automatically in this case suffices. This is a programmer error, after all, and the programmer can simply look at the stack trace.

Reply:
Interestingly enough I originally had a non-null assertion there but I thought you'd want me to handle the error properly 😆